armi/reactor/composites.py

```diff
@@ -1249,12 +1249,14 @@ def getNuclideNumberDensities(self, nucNames):
             # there are no children so no volume or number density
             return [0.0] * len(nucNames)
 
-        nucDensForEachComp = numpy.array(
-            [
-                [c.getNumberDensity(nuc) for nuc in nucNames]
-                for c in self.iterComponents()
-            ]
-        )  # c x n
+        densListForEachComp = []
+        for c in self.iterComponents():
+            numberDensityDict = c.getNumberDensities()
```

Comment:
This is faster because of using `getNumberDensities()`, right?

Reply:
Yes. This is small compared to some of the other changes, but `composites.getNumberDensity()` calls 

* `getNuclideNumberDensities()` if the the `composite` is not a `component`, which then calls `getNumberDensities()` on the child components
*  `self.p.numberDensities.get(nucName, 0.0)` if it is a `component`


https://github.com/terrapower/armi/blob/15a710f0620ca9521370f11b828de352f7b939e4/armi/reactor/composites.py#L1234-L1235